credentials/endpointcreds/internal/client/client.go

```diff
@@ -0,0 +1,147 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	"github.com/awslabs/smithy-go"
+	smithymiddleware "github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+)
+
+// HTTPClient is a client for sending HTTP requests
+type HTTPClient interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+// Options is the endpoint client configurable options
+type Options struct {
+	// The endpoint to retrieve credentials from
+	Endpoint string
+
+	// The authorization token to pass in the Authorization header with the request
+	AuthorizationToken string
```

Comment:
What do you think about the authorization value being a parameter of the operation input parameter instead of on the client?  Mainly suggesting this to separate state from the client, which is generally stateless.   Though I guess this is similar to the sdk's credentials.

Reply:
Yeah I was treating this similar to the authentication scheme's credentials which is usually stored on the client.